### PR TITLE
Modal form

### PR DESCRIPTION
**Pull request by mpace965**
Completes a lot, but not all, of [this card](https://trello.com/c/0Y2s1Ydo/10-user-sees-modal-form-for-adding-songs-when-joining-a-session)

Splitting it up into separate pull requests because its a lot of code.

* Adds [React Bootstrap](https://react-bootstrap.github.io/) for the Modal. This makes it so that we don't have to include bootstrap JavaScript. React Bootstrap reimplements that functionality with React for us.
* Clicking the x on the modal navigates you back to the home page.
* Themes that modal
* Adds [React DnD](http://react-dnd.github.io/react-dnd/) (not Dungeons and Dragons, unfortunately) to accomplish drag and drop.
* Implements the Contribution List using reactdnd
* Themes the Contribution List

all valid
some invalid

### CODE REVIEW

File: src/client/assets/components/.eslintrc.js [hunk 1]
```diff
@@ -1,10 +1,19 @@
...
   env: {
     browser: true,
   },
+  rules: {
+    "comma-dangle": ["error", {
+      "arrays": "only-multiline",
+      "objects": "only-multiline",
+      "imports": "only-multiline",
+      "exports": "only-multiline",
+      "functions": "ignore"
```
Comment (mpace965) [inline src/client/assets/components/.eslintrc.js]: Was getting some weird linter errors without this.

File: src/client/views/home.hbs [hunk 2]
```diff
@@ -1,6 +1,6 @@
 
 <div class='page-header site-title'>
-  <h1>{{title}}</h1>
+  <h1>Sessions</h1>
```
Comment (mpace965) [inline src/client/views/home.hbs:3]: This didn't actually need to be templated, as its static. Probably left over from when we first set up Handlebars.

File: webpack.config.js [hunk 3]
```diff
@@ -41,6 +41,9 @@ module.exports = {
       },
     ],
   },
+  resolve: {
+    extensions: ['.js', '.jsx'],
```
Comment (mpace965) [inline webpack.config.js:45]: Allows us to omit the '.jsx' or '.js' at the end of an `import` statement in modules transpiled with Webpack

**Review by mattgfisch — APPROVED** [review]
Really well done integrating the drag and drop contribution list. On second look this is a massive story and it definitely makes sense for it to be broken up into multiple parts. The styling of the contribution list will change as the story is completed, but that's easy and the main idea was to have the core functionality anyway. ⛵️

**Pull request merged**